src/assets/TokenRatesController.ts

```diff
@@ -192,6 +207,7 @@ export class TokenRatesController extends BaseController<
     });
     onNetworkStateChange(({ provider }) => {
       const { chainId } = provider;
+      this.update({ contractExchangeRates: {} });
```

Comment:
It would be great to add a test for this too! Ensuring that the prices are cleared when switching networks is pretty important for keeping users safe.

Reply:
added in [aaad27](https://github.com/MetaMask/controllers/pull/585/commits/aaad27132ff95bda984688713b5b87303afe3b88)